server/src/main/java/org/elasticsearch/index/mapper/vectors/DenseVectorFieldMapper.java

```diff
@@ -571,9 +572,31 @@ Function<StringBuilder, StringBuilder> errorByteElementsAppender(byte[] vector)
     );
 
     enum VectorSimilarity {
-        L2_NORM(VectorSimilarityFunction.EUCLIDEAN),
-        COSINE(VectorSimilarityFunction.COSINE),
-        DOT_PRODUCT(VectorSimilarityFunction.DOT_PRODUCT);
+        L2_NORM(VectorSimilarityFunction.EUCLIDEAN) {
```

Comment:
> As an aside, I was sort of expecting these operations to be encapsulated by some Lucene class, so you don't rely on this matching how the knn query scores.

It will be eventually.

Right now Lucene simply gives us score information from two vectors. It does not have mechanisms to transform the similarity to the final document score.